uiowa/uiowa
# Feature: tags in content types

# Summary 
Resolves https://github.com/uiowa/uiowa/issues/4480.

(this is @quamsta, adding to this PR a bit for clarity)

This PR allows both tags and related content to be shown as a view block in the pre-footer region in the following content types:

- Page
- Article
- Person

Other content types currently do not have the ability to display tags and related content. That may have to be done during a different slice.

Screenshots of the view block in action (**note: tag button styles may not be finalized!**):

- [Screenshot of tags and related content on a page](https://user-images.githubusercontent.com/472923/156051972-52eb05eb-3ebe-47ad-848f-c9b36722a1fb.png)
- [Screenshot of only tags on a page](https://user-images.githubusercontent.com/472923/156051982-aa656cd4-8322-43ee-84ae-b2fa0589dd37.png)
- [Screenshot of only related content on a page](https://user-images.githubusercontent.com/472923/156051991-639eacae-682a-4f52-8100-e845523a3258.png)

Each content type is configurable via their respective settings pages in 

-  `admin/config/sitenow/uiowa-pages` ([Screenshot](https://user-images.githubusercontent.com/472923/156046333-1393ec87-fe1b-4973-9bd2-248164f7d40c.png))
- `admin/config/sitenow/uiowa-articles`([Screenshot](https://user-images.githubusercontent.com/472923/156046344-9bfe460f-820f-4c17-baf3-cb3e7f9e9c71.png))
- `admin/config/sitenow/uiowa-people`([Screenshot](https://user-images.githubusercontent.com/472923/156046357-baf1f1df-1171-4a30-8d03-53288669498d.png))

# How to test

1.  Ensure you've tagged some pages, articles, or people
2. Pull down this branch
3. `drush @default.local csim`
4. Enable the desired tag display via the config options in SiteNow Pages, Articles, and/or People (see summary above)
5. Visit the pages you tagged in step 1 to see those tags.

# To do: 
- [x] Exclude currently visited page from related content
- [x] Resolve issue with uiowa_people.settings being reverted during a `drush csim`
- [x] Tag style finalization
- [x] Other things @bspeare had in mind for this PR 
- [x] Decouple tags and related content functionality
